source/javascripts/app/_search.js

```diff
@@ -29,22 +29,36 @@
     });
   }
 
+  var wait = (
+      function(){
+        var timeout_handle = 0;
+        return function(executing_function, wait_time){
+          clearTimeout (timeout_handle);
+          timeout_handle = setTimeout(executing_function, wait_time);
+        };
+      })();
+
```

Comment:
We already have a debounce function in `_toc.js`, over [here](https://github.com/lord/slate/blob/39aaa0fca567f56724b387fd109a37c8dfb7b59b/source/javascripts/app/_toc.js#L6-L17). Maybe we can just reuse that by moving it into a common usable place (like in `lib`) instead of implementing it twice? Not sure the functionality is exactly the same, but maybe it would be close enough? LMK what you think!